### ISSUE TITLE
facebook messenger inaccessibility

### ISSUE THREAD

**Issue by lallerke1** [nvaccess/nvda#9656]
Steps to reproduce:
1. Start Mozilla Firefox or Google Chrome.
2. Sign into www.messneger.com
3. Navigate to the active contacts.
4. Read the list with the arrow keys.

Actual behavior:
NVDA doesn't read the list neither with the arrow keys, nor with the tab key, even brows mode is activated with nvda+space.
Expected behavior:
Hotkeys should read the contact list in  messenger in firefox and chrome.
System configuration
Windows 10 Pro Version 1903 and 1807 x64
NVDA installed/portable/running from source:
installed
#### NVDA version:
alpha-17384,f20e6821
Windows version:
Windows 10 professional 1903 and 1809

Name and version of other software in use when reproducing the issue:
Mozilla Firefox 67.0, and google chrome 74.0.3729.169

Other information about your system:
Intel Celeron dualcore 4 GHZ

Other questions
Does the issue still occur after restarting your PC?
Yes, it does
Have you tried any other versions of NVDA? If so, please report their behaviors.
No, I did not.

Description:
NVDA doesn't the list view of the active contacts on the website messenger.com
This view includes ony mouse, and not keyboard rooted events, and is written in ajax, which isn't accessible for screen readers, and NVDA detects it as a Geckoi2Accessible edit field, as shown in the log file.
I attach to my request the nvda log, which was created on a hungarian win10 machine, with hungarian localization.
I also attach the source window of the elemend, which w	as provided by firefox and chroom if you select the view source context menu item.
[nvda.log](https://github.com/nvaccess/nvda/files/3241326/nvda.log)
[sourcecode.txt](https://github.com/nvaccess/nvda/files/3241329/sourcecode.txt)

**Comment by lallerke1** [comment]
Sorry, write not well the website:
www.messenger.com

**Comment by lallerke1** [comment]
@Qchristensen please visualize it, if you can.

**Comment by leonardder** [comment]
Do you have a reason to think that this is caused by NVDA, or whether it is an issue in the Facebook web app itself?

**Comment by Qchristensen** [comment]
Facebook messenger displays a list of contacts down the left hand side of the screen, the currently selected conversation in the middle, and details and options for the current contact on the right.  The focus starts in the "Type a message" edit below the currently selected conversation in the middle of the screen.  There might be a quicker shortcut to get directly to your list of contacts, but I don't use the messenger site much, so I tab once to get out of the edit box and go out of focus mode, then control+home to the top of the page, tab to the "search messages" then you can either start typing someone's name, or leave it blank, then down arrow to move through the list of contacts (which match what you typed, or all contacts).

In both Chrome and Firefox, the "focus" appears to move down through the contacts, in that the background behind the selected contact becomes shaded in Grey.  NVDA recognises that something has happened, but re-reads the contents of the search edit box.

Trying with Narrator on Windows 10, in Chrome, the focus jumps out of the search edit, and slowly moves down the list of current conversations (with NVDA, you could get to that by pressing TAB twice out of the search edit then using down arrow).  In Firefox with Narrator, press down arrow from the search edit does move through the list of suggested contacts, however it doesn't read anything as down arrow is pressed.

(Note, if you press TAB past the search edit and down arrow through the list of current conversations, NVDA will read that in both Firefox and Chrome.

CC @jcsteh who may be able to comment on what is happening in Firefox at least.

**Comment by jcsteh** [comment]
> The focus starts in the "Type a message" edit below the currently selected conversation in the middle of the screen. There might be a quicker shortcut to get directly to your list of contacts,

You can switch to browse mode and press control+home then t to jump to the first table, which is the contact list. You can then read the contact list in browse mode.

> tab to the "search messages" then you can either start typing someone's name, or leave it blank, then down arrow to move through the list of contacts (which match what you typed, or all contacts).
> In both Chrome and Firefox, the "focus" appears to move down through the contacts, in that the background behind the selected contact becomes shaded in Grey.

There is no "accessible" indication of this focus. Messenger is not using DOM focus, aria-activedescendant or even aria-selected.

> Trying with Narrator on Windows 10, in Chrome, the focus jumps out of the search edit, and slowly moves down the list of current conversations (with NVDA, you could get to that by pressing TAB twice out of the search edit then using down arrow).

My guess is that Narrator scan mode is engaging when you press down arrow, similar to NVDA's "Automatic focus mode for caret movement" browse mode setting (but in NVDA's case, that's disabled by default).

This issue isn't super specific - it's not clear whether we're talking about search, focus mode, browse mode, etc. - but certainly, I can interact with the contact list in browse mode. If we're talking about the search autocomplete focus issues, that's something which needs to be fixed by Facebook.

Closing as works for me.
